lib/using_restricted_array.rb

```diff
@@ -6,56 +6,111 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: linear or O(n), where n is the number of items in the array or the length of the input array
+# Space complexity: O(1)
 def length(array)
```

Comment:
Looks great, I also like how you're adding in additional details on time complexity